app/src/main/java/com/pokemonshowdown/data/ItemDex.java

```diff
@@ -3,83 +3,47 @@
 import android.content.Context;
 import android.util.Log;
 
+import com.paypal.android.sdk.D;
 import com.pokemonshowdown.app.R;
 import com.pokemonshowdown.application.MyApplication;
 
 import org.json.JSONException;
 import org.json.JSONObject;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.util.HashMap;
-import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 public class ItemDex {
     private final static String ITAG = ItemDex.class.getName();
-    private static ItemDex sItemDex;
-    private HashMap<String, String> mItemDexEntries;
+    public final static JSONObject DUMMY_OBJECT;
 
-    private ItemDex(Context appContext) {
-        mItemDexEntries = readFile(appContext);
-    }
-
-    private HashMap<String, String> readFile(Context appContext) {
-        HashMap<String, String> ItemDexEntries = new HashMap<>();
-        String jsonString;
+    static {
         try {
-            InputStream inputStream = appContext.getResources().openRawResource(R.raw.item);
-            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
-            StringBuilder stringBuilder = new StringBuilder();
-
-            String line;
-            while ((line = bufferedReader.readLine()) != null) {
-                stringBuilder.append(line);
-                stringBuilder.append("\n");
-            }
-            jsonString = stringBuilder.toString();
-            inputStream.close();
-
-            JSONObject jsonObject = new JSONObject(jsonString);
-
-            Iterator<String> keys = jsonObject.keys();
-
-            while (keys.hasNext()) {
-                String key = keys.next();
-                JSONObject entry = jsonObject.getJSONObject(key);
-                ItemDexEntries.put(key, entry.toString());
-            }
+            DUMMY_OBJECT = new JSONObject("{\"gen\":1,\"spritenum\":0,\"num\":0,\"name\":\"\",\"id\":\"dummy\",\"desc\":\"\"}");
         } catch (JSONException e) {
-            Log.d(ITAG, "JSON Exception");
-        } catch (IOException e) {
-            Log.d(ITAG, "Input Output problem");
-        }
-
-        return ItemDexEntries;
-    }
-
-    public static ItemDex get(Context c) {
-        if (sItemDex == null) {
-            sItemDex = new ItemDex(c.getApplicationContext());
+            // Will not happen, but is needed because it's final.
+            throw new Error();
         }
-        return sItemDex;
     }
 
     public static int getItemIcon(Context appContext, String itemName) {
+        itemName = itemName.replaceFirst("itemdex_", "");
```

Comment:
By the way: do we have an dummy icon for fail-safe purposes? If not, I will just use the pokeball_none-drawable.